pkg/liqonet/utils.go

```diff
@@ -117,13 +121,27 @@ func VerifyNoOverlap(subnets map[string]*net.IPNet, newNet *net.IPNet) bool {
 	return false
 }
 
-func GetClusterID(client *kubernetes.Clientset, cmName, namespace string) (string, error) {
+func GetClusterID(client *kubernetes.Clientset, cmName, namespace string, backoff wait.Backoff) (string, error) {
 	cmClient := client.CoreV1().ConfigMaps(namespace)
-	cm, err := cmClient.Get(context.TODO(), cmName, metav1.GetOptions{})
-	if err != nil {
-		return "", err
+	var cm *corev1.ConfigMap
+	var err error
+
+	notFound := func(error) bool {
+		return k8serrors.IsNotFound(err)
 	}
+
+	klog.Info("Getting ClusterID from ConfigMap...")
+	retryErr := retry.OnError(backoff, notFound, func() error {
+		klog.V(4).Info("Trying to get ConfigMap resource.")
```

Comment:
I would move this log message in the notFound handler by logging something like "error while getting ClusterID configmap, retrying..."